### ISSUE TITLE
[question] 'nmake' is not recognized as an internal or external command when trying to build conan-openssl

### ISSUE THREAD

**Issue by michaelmaguire** [conan-community/community#49]
### Description of Problem, Request, or Question

I'm trying to use the Conan recipe from https://github.com/conan-community/conan-boost to build OpenSSL.

What I find is that I get error:

```
E:\DevelopBp\mmaguir1\src\nx\conan-openssl>conan create . PORT/stable -s build_type=Debug
[...]
C:\Users\mmaguir1\.conan\data\openssl\1.0.2p\PORT\stable\build\5e08083bcde36dca41ed981ba123170d62070888\openssl-1.0.2p>perl util\mkdef.pl 32 ssleay  1>ms\ssleay32.def
WARN: replace_in_file didn't find pattern '/MD ' in './openssl-1.0.2p/ms/ntdll.mak' file.
WARN: replace_in_file didn't find pattern '/MT ' in './openssl-1.0.2p/ms/nt.mak' file.
openssl/1.0.2p@PORT/stable: WARN: .
openssl/1.0.2p@PORT/stable: WARN: ----------MAKE OPENSSL 1.0.2p-------------
'nmake' is not recognized as an internal or external command,
operable program or batch file.
openssl/1.0.2p@PORT/stable:
openssl/1.0.2p@PORT/stable: ERROR: Package '5e08083bcde36dca41ed981ba123170d62070888' build failed
openssl/1.0.2p@PORT/stable: WARN: Build folder C:\Users\mmaguir1\.conan\data\openssl\1.0.2p\PORT\stable\build\5e08083bcde36dca41ed981ba123170d62070888
ERROR: openssl/1.0.2p@PORT/stable: Error in build() method, line 193
        windows_make(config_options_string)
while calling 'windows_make', line 181
        run_in_src(make_command)
while calling 'run_in_src', line 123
        self.run("cd openssl-%s && %s" % (self.version, command))
        ConanException: Error 1 while executing cd openssl-1.0.2p && nmake -f ms\nt.mak
```

If I run e.g. ```vcvars64.bat```  from my VC directory:

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64>vcvars64
```

Now I can find nmake:

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64>nmake

Microsoft (R) Program Maintenance Utility Version 14.00.24210.0
Copyright (C) Microsoft Corporation.  All rights reserved.

NMAKE : fatal error U1064: MAKEFILE not found and no target specified
Stop.
```

And the conan build proceeds.

My question is this: in none of the other Conan recipe's I've used so far did I need to run vcvars64.bat -- what's special/broken about this recipe that it requires that I do so?

Is it the fact that it's not CMake-based?

### Package Details (Include if Applicable)
* Package Name/Version: **openssl/1.0.2p**
* Operating System: **Windows Server 2012 R2**
* Compiler+version: **Visual Studio 14 2015 v140_xp**
* Docker image: **N/A**

### Steps to reproduce (Include if Applicable)

Moved from https://github.com/conan-io/conan/issues/3665

**Comment by SSE4** [comment]
yes, I believe recipe needs small addition - usage of `tools.vcvars`, so nmake is automatically discovered. I'll add required changes and let you know.

**Comment by michaelmaguire** [comment]
I made a mistake in my problem description (now fixed) -- I had atttempted to use branch https://github.com/conan-community/conan-openssl/tree/release/1.0.2e which had this problem.

I later tried updating that branch by changing the version string to "1.0.2p" and still had this problem, but https://github.com/conan-community/conan-openssl/commits/release/1.0.2p does not.

( I happened to grab at commit https://bbgithub.dev.bloomberg.com/nx/conan-openssl/commit/321c807d9629a808ef895fd7df3c38e3134eccdc )

I was able to build both conan-libcurl and conan-grpc referencing this conan-openssl package without needing to call vcvars64.bat first.

Thanks and sorry for the noise.
